fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/SearchFilter.java

```diff
@@ -42,6 +42,22 @@
         QueryType(String type) {
             this.type = type;
         }
+
+        /**
+         * Get the QueryType enum from it's search type.
+         *
+         * @param type  The type of query (i.e. "insensitive_contains").
+         *
+         * @return the enum QueryType.
+         */
+        public static QueryType fromType(String type) {
```

Comment:
I would argue that this should be returning an `Optional` instead of a `null`, or throwing an `IllegalArgumentException` if the `type` is not a recognized `QueryType`. 

Returning an Optional makes it clear to people using the code that they may not get a QueryType back, and need to deal with that case. This makes sense if it's totally valid to have a string type that cannot be translated to a QueryType. 

The second case assumes that it's an error if we get an unrecognized type.

I'm assuming that it's an error to receive an unrecognized type, but I don't know yet what it's meant to be used for, so I can't say for certain.